fflogsapi/util/filters.py

```diff
@@ -20,6 +36,8 @@ def construct_filter_string(filters: dict[str, Any]) -> str:
         elif type(f) is bool:
             # bool type must be lowercase
             filter = f'{key}: {str(f).lower()}'
+        elif type(f) is dict:
+            filter = f'{key}: "{construct_filter_expression_string(f)}"'  
```

Comment:
While this is a more user friendly method of specifying filter expressions, they are a fflogs feature, not a GQL one. This is a function for constructing GQL filters.

Ideally, to be in line with future ambitions to have the package API mimic the GQL schema more closely, this should be a parameter on the function (e.g. `fight.events(filter_expression={...})`).

But for now, can we move this logic to `FFLogsFight._prepare_data_filters` and confine it apply to `filterExpression` only?